nav2_smac_planner/include/nav2_smac_planner/utils.hpp

```diff
@@ -154,6 +157,66 @@ inline void fromJsonToMotionPrimitive(
   }
 }
 
+/**
+ * @brief transform footprint into edges
+ * @param[in] robot position , orientation and  footprint
+ * @param[out] robot footprint edges
+ */
+inline std::vector<geometry_msgs::msg::Point> transformFootprintToEdges(
+  const double x, const double y, const double yaw,
+  const std::vector<geometry_msgs::msg::Point> & footprint)
```

Comment:
I don't think this place needs to be modified because even if it is modified to the std:: pair method, it still needs to be disassembled and used later

Reply:
True, but its for semantic understandability. We could also have it just be a vector of xy points `[x1 y1 x2 y2 x3 y3 ...]` or `[x1 x2 x3 y1 y2 y3 ...]`. There are infinite ways to structure the data, some which are generally understandable to anyone expecting `edges` to have a pair of nodes - some which are not.

Reply:
Does that make sense?